framework/src/media/InputHandler.cpp

```diff
@@ -200,14 +208,21 @@ void *InputHandler::workerMain(void *arg)
 			break;
 		}
 
-		auto size = stream->mBufferWriter->sizeOfSpace();
+		auto size = stream->sizeOfSpace();
 		if (size > 0) {
-			auto buf = new unsigned char[size];
-			if (worker->read(buf, size) <= 0) {
-				// Error occurred, or inputting finished
-				stream->mBufferWriter->setEndOfStream();
-				delete[] buf;
-				break;
+			unsigned char *buf;
+			if (stream->mPreloadData) {
+				size = stream->mPreloadLength;
+				buf = stream->mPreloadData;
+				stream->mPreloadData = NULL;
+			} else {
+				buf = new unsigned char[size];
+				if ((size = worker->read(buf, size)) <= 0) {
```

Comment:
size should be same as return of worker->read or it is acceptable if it is greater than 0?

Reply:
the return size greater than 0 is acceptable. e.g. when the file data source reach the end of file, we may get only part of data of desired size.